src/entities/activity/ui/ActivityCard/index.tsx

```diff
@@ -0,0 +1,117 @@
+  isPublic: boolean
+  onActivityCardClick: () => void
+}
+
+export const ActivityCard = ({ activityListItem, onActivityCardClick, isPublic }: ActivityCardProps) => {
+  const { lessThanSM } = useCustomMediaQuery()
+
+  const { isLoading, activity } = useActivity({
+    activityId: activityListItem.activityId,
+    isPublic,
+  })
+
+  const { activityEvents, progress } = useActivityEventProgressState({
+    activityId: activityListItem.activityId,
+    eventId: activityListItem.eventId,
+  })
+
+  const getCompletedActivitiesFromPosition = (position: number) => {
+    return position - 1
+  }
+
+  const isStatusScheduled = activityListItem.status === ActivityStatus.Scheduled
+
+  const isDisabled = isStatusScheduled
+
+  const isFlow = Boolean(activityListItem.flowId)
+
+  const isActivityInProgress = activityListItem.status === ActivityStatus.InProgress
+
+  const countOfCompletedQuestions = activityEvents.filter(item => item.answer.length).length
+
+  const activityLength = activity?.items.length || 0
+
+  const numberOfActivitiesInFlow = activityListItem.activityFlowDetails?.numberOfActivitiesInFlow || 0
+
+  const countOfCompletedActivities = getCompletedActivitiesFromPosition(
+    activityListItem.activityFlowDetails?.activityPositionInFlow || 0,
+  )
+
+  const flowProgress = (countOfCompletedActivities / numberOfActivitiesInFlow) * 100
+
+  const isSupportedActivity = useMemo(() => {
+    return activityBuilder.isSupportedActivity(activity)
+  }, [activity])
+
+  const onActivityCardClickHandler = () => {
+    if (!isSupportedActivity || isDisabled) {
+      return
+    }
+
+    return onActivityCardClick()
+  }
+
+  if (isLoading) {
+    return (
+      <ActivityCardBase>
+        <div className="activity-data">
+          <Loader />
+        </div>
+      </ActivityCardBase>
+    )
+  }
+
+  return (
+    <ActivityCardBase onClick={onActivityCardClickHandler} isDisabled={isDisabled}>
+      <Box
+        display="flex"
+        flex={1}
+        gap={lessThanSM ? "8px" : "24px"}
+        flexDirection={lessThanSM ? "column" : "row"}
+        data-testid="activity-card-content-wrapper"
+        sx={{ textTransform: "none" }}>
+        <ActivityCardIcon src={activityListItem.image} isFlow={Boolean(activityListItem.flowId)} />
+
+        <Box display="flex" flex={1} justifyContent="center" alignItems="flex-start" flexDirection="column" gap="8px">
+          <ActivityCardTitle title={activityListItem.name} />
+
+          {isActivityInProgress && <ActivityCardProgressBar percentage={isFlow ? flowProgress : progress} />}
+
+          <ActivityLabel
+            isFlow={isFlow}
+            activityLength={activityLength}
+            isSupportedActivity={isSupportedActivity}
+            isActivityInProgress={isActivityInProgress}
+            countOfCompletedQuestions={countOfCompletedQuestions}
+            countOfCompletedActivities={countOfCompletedActivities}
+            numberOfActivitiesInFlow={numberOfActivitiesInFlow}
```

Comment:
Can we work out the `countOfCompletedQuestions`, `countOfCompletedActivities`, `numberOfActivitiesInFlow`, `isActivityInProgress`, `isSupportedActivity` values in the ActivityLabel component? Or inside it's subcomponents

Reply:
Yes, we can but it will be duplicates of it. A lot of this values we also need in parent component